unintended_ml_bias/model_bias_analysis.py

```diff
@@ -103,6 +104,51 @@ def model_family_name(model_names):
   return prefix.strip('_')
 
 
+def normalized_mwu(data1, data2, model_name):
+  """Returns the number of pairs where the datapoint in data1 has a greater score than that from data2.""" 
+  scores_1 = data1[model_name]
+  scores_2 = data2[model_name]
+  n1 = len(scores_1)
+  n2 = len(scores_2)
+  u, _ = stats.mannwhitneyu(scores_1, scores_2, alternative = 'less')
+  return u/(n1*n2)
+
+
+def compute_within_negative_label_mwu(df, subgroup, label, model_name):
+  u_negative = normalized_mwu(df[~df[subgroup] & ~df[label]],
```

Comment:
I'm a bit confused about how df[subgroup] and df[label] both return boolean values that you can then pass into df again to get the correct rows. Could you explain in a bit more depth how this works? Happy to talk about this face to face if that would be easier.